Update containers API version from 2024-05-01 to 2024-09-01

tiesmaster opened the pull request:
<!--  All Submissions -->


## Community Note
<!-- Please leave the community note as is. -->
* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description

This is a precursor PR for issue #27243, to update the `.../containerservice/{2024-05-01 -> 2024-09-01}/*` API. This is to prepare to add support for controlling AKS [Advanced Container Networking Services](https://learn.microsoft.com/en-us/azure/aks/advanced-container-networking-services-overview).

I've ran the unittest suite, and the following acceptance test: `TestAccKubernetesCluster_runCommand`, and all passed.

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

---

❗ As this is a version bump of the one of the APIs, the rest of the sections are not relevant, and hence removed.

ms-henglu commented:
![image](https://github.com/user-attachments/assets/4c753ad2-b8b0-4de8-a9b6-b9eb9a735879)

tiesmaster commented:
> The failed tests seem unrelated with this change.

@ms-henglu Indeed, they don't seem related, I trust you on that ;) Is this a private Teamcity instance, as I don't see these in the list of checks?

Is there anything I can do to move this PR forward?

ms-henglu commented:
Yes, it's a private team city. The Hashicorp reviewers will review and merge this PR.

sf-msft commented:
@tiesmaster What does the delta between 2024-09-01 and 2024-10-01 look like? If review cycles are long, might as well get it to the latest

stephybun reviewed (changes requested):
@tiesmaster apologies for the delay in review here and thanks for running the tests on this PR @ms-henglu. Would you mind resolving the merge conflicts? Once that's done this should be good to go.

tiesmaster commented:
@stephybun No worries, thanks for getting to the PR. Looking at the changes, and merge conflicts, it seems like there is a little bit more needed than only resolving the merge conflict. It looks like an additional AKS API has been pulled in (`trustedaccess`) in the mean time, and I will need to upgrade that API as well (so also vendor the new version of that).

I will do that in 3 to 4 days (max a week).

stephybun commented:
@tiesmaster `trustedaccess` was an emergency upgrade since the preview version it was on was deprecated. You're welcome to take the time to upgrade this as well, but it would also be acceptable for you to leave the version for `trustedaccess` alone for the time being and it'll get bumped next time all the containerservice APIs get updated.

tiesmaster commented:
@stephybun I've rebased my changes, and fixed the merge conflicts. It was easy to also include the `trustedaccess` (didn't have any breaking changes), so I took to liberty to include that one as well.

I've ran the unit tests locally, without any issues, and I'm running a smoke test acceptance tests (though, that takes a while, ofc), if that comes back with problems, then I'll convert the PR to draft.

@ms-henglu Can you kick off the tests on your end again? Thank you!

tiesmaster commented:
@stephybun Might be good to see if we can move this to the next step, in order to prevent any new merge conflicts to popup.

@ms-henglu Did you manage to kick-off a new test run?

ms-henglu commented:
<img width="1134" alt="image" src="https://github.com/user-attachments/assets/bbc64e7c-888f-4b07-a295-e848eabb931d" />

stephybun reviewed (approved):
LGTM ⛰️

tiesmaster commented:
Awesome, I'm gonna start working on the main PR for issue #27243
